diskv.go

```diff
@@ -226,14 +224,27 @@ func (d *Diskv) createKeyFileWithLock(pathKey *PathKey) (*os.File, error) {
 
 // writeStream does no input validation checking.
 func (d *Diskv) writeStreamWithLock(pathKey *PathKey, r io.Reader, sync bool) error {
+	// fullPath is the on-disk location of the key
+	fullPath := d.completeFilename(pathKey)
+
 	if err := d.ensurePathWithLock(pathKey); err != nil {
 		return fmt.Errorf("ensure path: %s", err)
 	}
 
+	// createKeyFileWithLock gives us a temporary file we can write to.
+	// We'll move it when we're all done.
 	f, err := d.createKeyFileWithLock(pathKey)
 	if err != nil {
 		return fmt.Errorf("create key file: %s", err)
 	}
+	// In case something bad happens, we want to delete the temporary file,
+	// lest we leave junk in the store.
+	defer func() {
+		if r := recover(); r != nil {
+			os.Remove(f.Name())
+			panic(r)
+		}
+	}()
```

Comment:
No recovers, please.

Reply:
What's your preferred solution to remove the temporary file if the write fails half-way through? Without this, TestAtomicWrite will fail.

Reply:
Would the write fail with a panic, or with an error?

Reply:
In TestAtomicWrite, we call WriteStream using a special io.Reader which panics when you call Read. This code catches that panic, removes the temporary file, and then throws the panic further up again. It maintains the *existing* behavior.

Reply:
Eh. Fix the test.

Reply:
So, the reason I put in that defer/recover is that if there is a panic in the middle of `writeStreamWithLock`, the temporary file (key + random extension) still exists in the diskv storage directory, will never be cleaned up, and will appear as its own key if you list keys.

In thinking about it, I realized that even leaving aside the panic scenario, we have a problem: If you call `WriteStream` on a existing key named "foo" and give it an extremely slow io.Reader as the input, someone who calls `Keys` will see a key named "foo" and a key named "foo.03f7a83" or whatever random suffix we give it.

With this in mind, I'm actually considering if #63 was the more correct solution to the problem after all.

Reply:
What would cause a panic?

Reply:
We should ensure that the temporary file with the random suffix is not "visible" to Diskv normally.

Reply:
my recent changes ensure this.